### PR TITLE
Feature/player commands js

### PR DESCRIPTION
**Pull request by OttoRocket-LVP**
Thank you for making a contribution to [Las Venturas Playground](https://sa-mp.nl/)!

Please take a moment to fill in the questions in this template, to make it easier for our developers
to understand what you're doing.

## What is being changed?
_Allow the usage of /my within JavaScript. (Redirect to Pawn if the subcommand is not known by JS.)
Allow players to use /my spawnweapons to buy spawnweapons._

## Why is this being changed?
_To improve the QoL for players by allowing them to buy spawnweapons.
To prepare for the option to have persisting spawn weapons._

## How are you making the change?
  [X] I have a working check-out of Las Venturas Playground.
  [X] I have included tests for any changed JavaScript code.
  [X] I have tested this change myself on my local server.

### CODE REVIEW

**Review by Specifer — COMMENTED** [review]
Nice work man, it's cool af!

**Review by RussellLVP — COMMENTED** [review]
I really like the premise of this, and think this is a great and really impactful thing to do! Thanks :)

At a high level, I have a few thoughts:
  * Love splitting it up in file-per-command, like we do with the smaller commands in the `playground` feature.
  * We shouldn't flip around _command name_ and _player_ for the JavaScript-based commands. Your tests don't seem to do that, is the `PLAYER_COMMANDS_INVALID_COMMAND` message just wrong?
  * We should revise the data format for spawn weapons, but can do that after this lands. It's quite messy.
  * Probably need to remove the "p" command from Pawn as well?

**Comment by OttoRocket-LVP** [comment]
- I don't see why the `PLAYER_COMMANDS_INVALID_COMMAND` is wrong it is turned around as implemented? Honestly I turned it around to await your feedback on the matter due to not geting `/p [id] [subcommand] [params] ` to work out of the box. (And thought `/p [subcommand] [id] [params]` didn't seem so bad. (Although might get confusing for admins). I agree I can put some more time in it to figure out how to get `/p [id] [subcommand] [params]` to work.
- I agree we should revise this. However I wanted to do this in smaller steps. I'm also still not 100% aware of the impact on the overall pawn if changing it thus want to take this in another step.
- Remove it from the `OnPlayerCommand`. I agree, I'm not sure as to why I didn't. (it should never reach that anyway)

File: data/messages.json [hunk 1]
```diff
@@ -448,6 +448,14 @@
   "NUWANI_ADMIN_KICKED": "%s (IRC) has kicked %s (Id:%d): %s",
   "NUWANI_PLAYER_BANNED_NOTICE": "{FF9900}You have been banned by %s for %d days{FFFFFF}: %s",
   "NUWANI_PLAYER_KICKED_NOTICE": "{FF9900}You have been kicked by %s{FFFFFF}: %s",
+  
+  "PLAYER_COMMANDS_INVALID_SPAWN_WEAPON" : "@error Sorry, id %d is not a valid spawn weapon.",
+  "PLAYER_COMMANDS_SPAWN_WEAPONS_TELEPORT" : "@error Sorry, you can't get weapons now because you %s.",
```
Comment (RussellLVP) [inline data/messages.json]: nit: alphabetize

File: data/server/callbacks.txt [hunk 2]
```diff
@@ -16,6 +16,9 @@ forward OnSetPlayerSkinId(playerId, skinId, bool: uponNextSpawn);
 forward OnGiveWeapon(playerId, weaponId, ammo);
 forward OnResetPlayerWeapons(playerId);
 forward OnRemovePlayerWeapon(playerId, weaponId);
+forward OnPlayerCommand(playerid, cmdtext[]);
+forward OnGiveSpawnArmour(playerId);
+forward OnGiveSpawnWeapon(playerId, weaponId, multiplier);
```
Comment (RussellLVP) [inline data/server/callbacks.txt]: Only JavaScript functions called by Pawn have to be listed here. You can call any public Pawn function from JavaScript. (As long as it starts with On*)

File: javascript/features/player_commands/commands/spawn_weapons.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+const ExtraPriceFactor = 1.5;
+
+// Command: spawnweapons [weaponId] [multipler]
+export default class SpawnWeapons extends PlayerCommand {
+    get name() { return 'spawnweapons'; }
+
+    build(commandBuilder) {
+        commandBuilder
+            .parameters([{ name: 'weapon', type: CommandBuilder.NUMBER_PARAMETER, optional: false },
+            { name: 'multiplier', type: CommandBuilder.NUMBER_PARAMETER, optional: false }])
```
Comment (RussellLVP) [inline javascript/features/player_commands/commands/spawn_weapons.js]: nit: `optional: false` is implied, elsewhere too

File: javascript/features/player_commands/commands/spawn_weapons.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            .build(SpawnWeapons.prototype.onSpawnWeaponsCommand.bind(this));
+    }
+
+    buildAdmin(commandBuilder) {
```
Comment (RussellLVP) [inline javascript/features/player_commands/commands/spawn_weapons.js:27]: Let's go with this for now, but I'll have a think about how we could remove the duplication.

File: javascript/features/player_commands/commands/spawn_weapons.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        if (multiplier <= 0 || multiplier > 100) {
+            player.sendMessage(Message.PLAYER_COMMANDS_SPAWN_WEAPONS_INVALID_AMOUNT, multiplier);
+            return;
+        }
+
+        const weaponData = WeaponData.getWeaponById(weapon);
+
+        if (player.id === subject.id) {
```
Comment (RussellLVP) [inline javascript/features/player_commands/commands/spawn_weapons.js]: nit: just compare the instances, much easier :)

```javascript
if (player === subject) {}
```

Note that you can just add the `Player` instances in Map, Set etc. too

File: javascript/features/player_commands/player_commands_commands.test.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+    it('should pawninvoke unknown p command', async assert => {
+        gunther.level = Player.LEVEL_ADMINISTRATOR;
+        assert.isTrue(await gunther.issueCommand('/p 1'));
+    });
+
+    it('should pawninvoke unknown p command', async assert => {
+        gunther.level = Player.LEVEL_ADMINISTRATOR;
+        assert.isTrue(await gunther.issueCommand('/p 1 test more test'));
+    });
```
Comment (RussellLVP) [inline javascript/features/player_commands/player_commands_commands.test.js]: What do these tests actually test?

They verify that the listener for the given command can be called, but not what it does (or if it does anything).

**Review by RussellLVP — DISMISSED** [review]
A few comments, thanks! This is ace, I'm excited to be able to move more stuff to JavaScript.

File: data/messages.json (hunk 1 above)
Comment (OttoRocket-LVP) [reply data/messages.json]: My bad, renamed a few didn't re-order. Sorry.

**Review by RussellLVP — APPROVED** [review]

**Pull request merged**